src/cogent3/app/__init__.py

```diff
@@ -16,31 +16,28 @@
 
 def _get_app_attr(name, obj, mod, is_composable):
     """returns app details for display"""
+    _types = {"_data_types": [], "_return_types": []}
 
-    _types = {"_input_types": [], "_output_types": [], "_data_types": []}
-
-    for tys in _types.keys():
+    for tys in _types:
         types = getattr(obj, tys, None) or []
         types = [types] if type(types) == str else types
         _types[tys] = [{None: ""}.get(e, e) for e in types]
 
-    row = [
+    return [
         mod.__name__,
         name,
         is_composable,
         obj.__doc__,
-        ", ".join(_types["_input_types"]),
-        ", ".join(_types["_output_types"]),
-        ", ".join(_types["_data_types"]),
+        _types["_data_types"],
+        _types["_return_types"],
     ]
-    return row
 
 
 def available_apps():
     """returns Table listing the available apps"""
     from cogent3.util.table import Table
 
-    from .composable import Composable, user_function
+    from .composable import Composable, is_composable, user_function
```

Comment:
First, I changed and used `__app_registry` dict instead of `is_composable` function. But when I see `hypothesis` class has a composable base class, I have to use `is_composable` function. If you want a special use of `__app_registry`, please tell me to change this code.

Reply:
Now I understand what is your purpose, directly using names from `__app_registry`. I changed it to use names directly in `__app_registry`.